flink-tests/src/test/java/org/apache/flink/test/util/TestUtils.java

```diff
@@ -94,6 +102,59 @@ public static void waitUntilJobInitializationFinished(
                 userCodeClassloader);
     }
 
+    public static CheckpointMetadata loadCheckpointMetadata(String savepointPath)
+            throws IOException {
+        CompletedCheckpointStorageLocation location =
+                AbstractFsCheckpointStorageAccess.resolveCheckpointPointer(savepointPath);
+
+        try (DataInputStream stream =
+                new DataInputStream(location.getMetadataHandle().openInputStream())) {
+            return Checkpoints.loadCheckpointMetadata(
+                    stream, Thread.currentThread().getContextClassLoader(), savepointPath);
+        }
+    }
+
+    public static List<Path> findAllSortedExternalizedCheckpoint(File checkpointDir, JobID jobId)
+            throws IOException {
+        return findAllExternalizedCheckpoint(checkpointDir, jobId).stream()
+                .sorted(
+                        Comparator.comparingInt(
+                                path ->
+                                        Integer.parseInt(
+                                                path.getFileName()
+                                                        .toString()
+                                                        .substring(
+                                                                CHECKPOINT_DIR_PREFIX.length()))))
+                .collect(Collectors.toList());
+    }
+
+    public static Optional<Path> findExternalizedCheckpoint(File checkpointDir, JobID jobId)
+            throws IOException {
+        return findAllExternalizedCheckpoint(checkpointDir, jobId).stream().findAny();
+    }
+
+    public static List<Path> findAllExternalizedCheckpoint(File checkpointDir, JobID jobId)
+            throws IOException {
+        try (Stream<Path> checkpoints =
+                Files.list(checkpointDir.toPath().resolve(jobId.toString()))) {
+            return checkpoints
+                    .filter(path -> path.getFileName().toString().startsWith(CHECKPOINT_DIR_PREFIX))
+                    .filter(
+                            path -> {
+                                try (Stream<Path> checkpointFiles = Files.list(path)) {
+                                    return checkpointFiles.anyMatch(
+                                            child ->
+                                                    child.getFileName()
+                                                            .toString()
+                                                            .equals(METADATA_FILE_NAME));
+                                } catch (IOException ignored) {
+                                    return false;
+                                }
```

Comment:
Could you explain why is the exception ignored here?
With externalized checkpoints, the checkpoint shouldn't be removed, should it?